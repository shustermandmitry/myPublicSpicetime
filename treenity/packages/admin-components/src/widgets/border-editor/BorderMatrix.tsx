/*
 * Copyright (c) 2024. Treenity Inc.
 */

import Button from '@/components/button';
import Icon from '@/components/icon';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { FC, useCallback } from 'react';

export type BorderMatrixItems = 'top' | 'right' | 'bottom' | 'left' | 'all';

interface BorderMatrixProps {
  value: BorderMatrixItems;
  onChange(value: BorderMatrixItems): void;
}

const BorderMatrix: FC<BorderMatrixProps> = ({ value = 'all', onChange }) => {
  const isSelected = useCallback(
    (name: BorderMatrixItems) => {
      if (value === name) {
        return 'primary';
      }

      return 'secondary-outlined';
    },
    [value],
  );

  return (
    <Root>
      <div>
        <ButtonStyled
          onClick={() => onChange('top')}
          type={isSelected('top')}
          size="x-small"
          icon={<Icon name="border-top_outlined" />}
        />
      </div>
      <div>
        <ButtonStyled
          onClick={() => onChange('left')}
          type={isSelected('left')}
          size="x-small"
          icon={<Icon name="border-left_outlined" />}
        />
        <ButtonStyled
          onClick={() => onChange('all')}
          type={isSelected('all')}
          size="x-small"
          icon={<Icon name="border-all_outlined" />}
        />
        <ButtonStyled
          onClick={() => onChange('right')}
          type={isSelected('right')}
          size="x-small"
          icon={<Icon name="border-right_outlined" />}
        />
      </div>
      <div>
        <ButtonStyled
          onClick={() => onChange('bottom')}
          type={isSelected('bottom')}
          size="x-small"
          icon={<Icon name="border-bottom_outlined" />}
        />
      </div>
    </Root>
  );
};

const ButtonStyled = styled(Button)`
  ${p =>
    p.type === 'secondary-outlined' &&
    css`
      i {
        color: ${p.theme.colorText};
      }
    `}
`;

const Root = styled.div`
  flex-direction: column;
  width: 64px;

  &,
  & > div {
    display: flex;
    gap: 2px;
    justify-content: center;
  }
`;

export default BorderMatrix;
