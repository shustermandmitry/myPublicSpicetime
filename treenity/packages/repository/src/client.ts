export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
