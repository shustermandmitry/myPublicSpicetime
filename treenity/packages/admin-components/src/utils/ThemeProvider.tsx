/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
