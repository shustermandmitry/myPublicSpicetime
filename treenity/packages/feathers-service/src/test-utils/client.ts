import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
