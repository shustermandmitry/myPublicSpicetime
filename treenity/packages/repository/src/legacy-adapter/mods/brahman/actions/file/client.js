import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
