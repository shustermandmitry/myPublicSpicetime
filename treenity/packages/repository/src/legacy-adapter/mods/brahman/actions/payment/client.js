import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
