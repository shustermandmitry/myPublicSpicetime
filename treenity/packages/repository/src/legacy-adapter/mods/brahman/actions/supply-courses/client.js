import './TelegramActionSupplyCourses.client';
import './TelegramCoursesOperator.client';
import './TelegramActionBecomeOperator.client';
