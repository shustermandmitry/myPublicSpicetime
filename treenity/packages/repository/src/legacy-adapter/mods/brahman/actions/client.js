import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
