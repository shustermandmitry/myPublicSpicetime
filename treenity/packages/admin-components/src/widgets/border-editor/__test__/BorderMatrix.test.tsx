/*
 * Copyright (c) 2024. Treenity Inc.
 */
import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
import BorderMatrix, { BorderMatrixItems } from '../BorderMatrix';

const cookies = createCookies(parseCookies(document.cookie), setCookie);
let valueMock: BorderMatrixItems = 'all';

const updateMock = jest.fn((value: BorderMatrixItems) => {
  valueMock = value;
  return value;
});

const defaultValue = 'all';

describe('BorderMatrix', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = 'all';
  });

  it('render BorderMatrix component', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <BorderMatrix onChange={updateMock} value={defaultValue} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-en83df"]`);
    const buttons = root!.querySelectorAll(`button[class^="ant-btn"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(buttons).toHaveLength(5);
    [...buttons].forEach((button, index) => {
      if (index === 2) {
        expect(button).toHaveClass('ant-btn-primary');
      } else {
        expect(button).not.toHaveClass('ant-btn-primary');
        expect(button).toHaveClass('ant-btn-default');
      }
    });
  });

  it('change BorderMatrix component', async () => {
    const { container, rerender } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <BorderMatrix onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-en83df"]`);
    const buttons = root!.querySelectorAll(`button[class^="ant-btn"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();

    expect(buttons).toHaveLength(5);
    [...buttons].forEach((button, index) => {
      if (index === 2) {
        expect(button).toHaveClass('ant-btn-primary');
      } else {
        expect(button).not.toHaveClass('ant-btn-primary');
        expect(button).toHaveClass('ant-btn-default');
      }
    });

    expect(valueMock).toBe('all');

    await waitFor(() => fireEvent.click(buttons[4]));

    rerender(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <BorderMatrix onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(valueMock).toBe('bottom');

    expect(updateMock).toHaveBeenCalledTimes(1);

    [...buttons].forEach((button, index) => {
      if (index === 4) {
        expect(button).toHaveClass('ant-btn-primary');
      } else {
        expect(button).not.toHaveClass('ant-btn-primary');
        expect(button).toHaveClass('ant-btn-default');
      }
    });
  });
});
