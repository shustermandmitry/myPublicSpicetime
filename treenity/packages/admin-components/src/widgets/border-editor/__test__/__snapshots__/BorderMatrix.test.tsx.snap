// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderMatrix change BorderMatrix component 1`] = `
.emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 64px;
}

.emotion-0,
.emotion-0>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  background: #FFF;
  color: #26233F;
  border-color: #E9E8EE;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-2>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-2.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-2 i {
  color: #27ae60;
}

.emotion-2.ant-btn-default:not(:disabled):not(.emotion-2.ant-btn-disabled):hover {
  background: #FFF;
  color: #47BA73;
  border-color: #E5E4EC;
}

.emotion-2.ant-btn-default:not(:disabled):not(.emotion-2.ant-btn-disabled):active {
  background: #FFF;
  color: #178749;
  border-color: #E1DFE8;
}

.emotion-2 i {
  font-size: 16px;
}

.emotion-2>span {
  gap: 4px;
}

.emotion-2,
.emotion-2 span {
  line-height: 10px;
}

.emotion-2.emotion-2>span>span>i {
  font-size: 12px;
}

.emotion-2 i {
  color: rgba(38, 35, 63, 0.88);
}

.emotion-3 [class^='icon-'],
.emotion-3 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-4 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-10 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  border: none;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-10>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-10.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-10 i {
  font-size: 16px;
}

.emotion-10>span {
  gap: 4px;
}

.emotion-10,
.emotion-10 span {
  line-height: 10px;
}

.emotion-10.emotion-10>span>span>i {
  font-size: 12px;
}

<div
  class="emotion-0"
>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-top_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-left_outlined emotion-4"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-1 ant-btn-primary ant-btn-sm ant-btn-icon-only emotion-10"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-all_outlined emotion-4"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-right_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-bottom_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`BorderMatrix render BorderMatrix component 1`] = `
.emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 64px;
}

.emotion-0,
.emotion-0>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  background: #FFF;
  color: #26233F;
  border-color: #E9E8EE;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-2>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-2.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-2 i {
  color: #27ae60;
}

.emotion-2.ant-btn-default:not(:disabled):not(.emotion-2.ant-btn-disabled):hover {
  background: #FFF;
  color: #47BA73;
  border-color: #E5E4EC;
}

.emotion-2.ant-btn-default:not(:disabled):not(.emotion-2.ant-btn-disabled):active {
  background: #FFF;
  color: #178749;
  border-color: #E1DFE8;
}

.emotion-2 i {
  font-size: 16px;
}

.emotion-2>span {
  gap: 4px;
}

.emotion-2,
.emotion-2 span {
  line-height: 10px;
}

.emotion-2.emotion-2>span>span>i {
  font-size: 12px;
}

.emotion-2 i {
  color: rgba(38, 35, 63, 0.88);
}

.emotion-3 [class^='icon-'],
.emotion-3 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-4 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-10 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  border: none;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-10>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-10.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-10 i {
  font-size: 16px;
}

.emotion-10>span {
  gap: 4px;
}

.emotion-10,
.emotion-10 span {
  line-height: 10px;
}

.emotion-10.emotion-10>span>span>i {
  font-size: 12px;
}

<div
  class="emotion-0"
>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-top_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-left_outlined emotion-4"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-1 ant-btn-primary ant-btn-sm ant-btn-icon-only emotion-10"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-all_outlined emotion-4"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-right_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
  <div>
    <button
      class="ant-btn emotion-1 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-2"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-3"
        >
          <i
            class="icon-border-bottom_outlined emotion-4"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;
